app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/apps/ManageAppsProtectionViewModel.kt

```diff
@@ -74,7 +79,19 @@ class ManageAppsProtectionViewModel @Inject constructor(
             .launchIn(viewModelScope)
     }
 
-    internal suspend fun getProtectedApps() = excludedApps.getAppsAndProtectionInfo().map { ViewState(it) }
+    internal suspend fun getProtectedApps(): Flow<ViewState> {
+        return excludedApps.getAppsAndProtectionInfo()
```

Comment:
This is accessing disk every single time, would it make sense to keep an in memory list? we avoid accessing IO each time we change the filter and it would be much faster. In my device the change from Protection Disabled to Protection Enabled takes half a second